docs/rst/container_yml/reference.rst

```diff
@@ -773,4 +900,70 @@ requested_storage        The amount of storage being requested. Defaults to 1Gi.
                          See `compute resources <http://kubernetes.io/docs/user-guide/compute-resources/>`_ for abbreviations.
 ======================== ========================================================================================================================
 
+.. _secrets_top:
+
+Secrets
+```````
+
+The top-level ``secrets`` directive, however, has been modified slightly. The following example
+``container.yml`` shows the service level ``secrets`` directive, and the new top-level ``secrets`` format:
+
+.. code-block:: yaml
+
+    version: '2'
+    services:
+      web:
+        from: centos:7
+        entrypoint: [/usr/bin/entrypoint.sh]
+        working_dir: /
+        user: apache
+        command: [/usr/bin/dumb-init, httpd, -DFOREGROUND]
+        ports:
+        - 8000:8080
+        - 4443:8443
+        roles:
+        - apache-container
+        secrets:
+        - apache-certs
+
+    secrets:
+      my-secrets:
+        docker: {}
+        openshift:
+          state: present
+          force: false
+          type: Generic
```

Comment:
Should support and generate any of the secret types. I can document the types or add a link to the official k8s documentation. Which do you think is better?